### PR TITLE
Update MerlinAU.sh

### PR DESCRIPTION
**Pull request by Firetop**
Changes Made:

**1. Conditional Deletion of ZIP File:** 
Modified the script to only delete the firmware ZIP file if it's saved locally (/home/root). 
If the ZIP file is on a USB drive (using a /mnt location), the script does not delete it.

**2. ROG Build Detection and Prompt:** 
Addressed an issue where the script was not prompting the user to select a build to flash when a ROG build was present in the directory.

**3. Switch to 'ROGBuild' Setting:** 
Changed references from 'local' to 'ROGBuild' in the settings file, updating the script to retrieve and handle the 'ROGBuild' line instead of 'local'.

**4. Handling 'ROGBuild' Setting Returning 'TBD':**
Modified the script to check the directory for a ROG build and prompt the user if the 'ROGBuild' property in the settings file returns 'TBD'.

**5. Fixing Retrieval of 'ROGBuild' Setting Value:** 
Corrected the Get_Custom_Setting function to properly retrieve the value of the 'ROGBuild' setting from the settings file when it's set.

**6. Modify change_build_type Function:** 
Altered the change_build_type function so that it no longer checks the directory for a ROG build before changing the build type. Instead, it checks the settings file for the 'ROGBuild' setting.

**7. Display 'Change Firmware Build Type' in Menu:** 
Updated the show_menu function to display the "Change Firmware Build Type" option if the 'ROGBuild' line exists in the settings file with a value of either 'y' or 'n'.

**8. Display Current Build Type in Menu:** 
Updated the show_menu function to display the current build type (either "ROG Build" or "Pure Build") alongside the option to change the build type.

**9. Move Logging Start Location** 
Updated the _RunFirmwareUpdateNow_() function to not include the checking of credentials in the logs. (Just reordered / Moved where the log starts)

### CODE REVIEW

**Comment by Firetop** [comment]
image

**Comment by Firetop** [comment]
> > I think this is ready to merge. But please feel free to review. I'll give it a few days before I push it through myself. It isn't a burning change for most people.
> 
> Sounds good. I'm just about to have dinner so I'll take a look after (if the movie we're going to watch tonight is not as good as I hope - Oppenheimer :>). Otherwise, tomorrow night.

Such a long movie! Took me 2 tries to finish it. 
But I did finally finish it, and enjoyed it. The more background you have on the history the better I feel. 

No rush! Chat soon and enjoy Martin! :)

File: MerlinAU.sh [hunk 1]
```diff
@@ -1652,59 +1652,63 @@ _RunFirmwareUpdateNow_()
...
+	if unzip -o "$FW_ZIP_FPATH" -d "$FW_BIN_DIR" -x README*
+	then
+		# Get the default USB mount point path
+		usbMountPoint="$(_GetDefaultUSBMountPoint_)"
+		if [ $? -eq 1 ]; then
+			Say "${REDct}**ERROR**${NOct}: Failed to get the default USB mount point path."
+			# Consider how to handle this error. For now, we'll not delete the ZIP file.
+		else
+			# Check if the ZIP file path is not under the USB mount directory
+			if ! echo "$FW_ZIP_FPATH" | grep -qE "^${usbMountPoint}"
```
Comment (Martinski4GitHub) [inline MerlinAU.sh:1665]: [Lines 1665-1669] This code results in deleting the ZIP file even when it was on a USB-attached drive (where Entware is installed). I've already got a fix for it.

**Comment by Martinski4GitHub** [comment]
> 
> Such a long movie! Took me 2 tries to finish it. But I did finally finish it, and enjoyed it. The more background you have on the history the better I feel.
> 
We ended up watching "Tenet" from the same directory. Awesome movie, but it requires paying careful attention to everything because it may turn out to be significant later on. It really requires a 2nd viewing to finally understand all the intricacies of the plot. Certainly not your average movie, and perhaps not for everyone.

File: MerlinAU.sh (hunk 1 above)
Comment (Firetop) [reply MerlinAU.sh:1665]: Crap. I was so sure it worked ! :(

Thanks for catching this Martin! This is why I need a second pair of eyes haha. 
2 brains is always better than one.

**Comment by Firetop** [comment]
> 
> and perhaps not for everyone.

I can agree to this line. 

Tenet was one of the few movies I didn't even try to finish lol! I got so lost trying to follow it... I ain't ashamed to say it.
For me, lots of the time I watch a movie to turn off my mind.

Tenet felt like it was trying to be complicated to follow on purpose, and honestly for a guy that watches movies as he's passing out.  It wasn't for me lol!

**Pull request merged**